pkg/api/api_test.go

```diff
@@ -45,3 +53,79 @@ func newTestServer(t *testing.T, o testServerOptions) *http.Client {
 		}),
 	}
 }
+
+func TestParseName(t *testing.T) {
+	const bzzHash = "89c17d0d8018a19057314aa035e61c9d23c47581a61dd3a79a7839692c617e4d"
+
+	testCases := []struct {
+		desc       string
+		name       string
+		log        logging.Logger
+		res        resolver.Interface
+		noResolver bool
+		wantAdr    swarm.Address
+		wantErr    error
+	}{
+		{
+			desc:    "empty name",
+			name:    "",
+			wantErr: api.ErrInvalidChunkAddress,
+		},
+		{
+			desc:    "bzz hash",
+			name:    bzzHash,
+			wantAdr: swarm.MustParseHexAddress(bzzHash),
+		},
+		{
+			desc:       "no resolver connected with bzz hash",
+			name:       bzzHash,
+			noResolver: true,
+			wantAdr:    swarm.MustParseHexAddress(bzzHash),
+		},
+		{
+			desc:       "no resolver connected with name",
+			name:       "itdoesntmatter.eth",
+			noResolver: true,
+			wantErr:    api.ErrNoResolver,
+		},
+		{
+			desc: "name not resolved",
+			name: "not.good",
+			res: resolverMock.NewResolver(
+				resolverMock.WithResolveFunc(func(string) (swarm.Address, error) {
+					return swarm.ZeroAddress, errors.New("failed to resolve")
+				}),
+			),
+			wantErr: api.ErrInvalidChunkAddress,
+		},
+		{
+			desc:    "name resolved",
+			name:    "everything.okay",
+			wantAdr: swarm.MustParseHexAddress("89c17d0d8018a19057314aa035e61c9d23c47581a61dd3a79a7839692c617e4d"),
+		},
+	}
+	for _, tC := range testCases {
+		if tC.log == nil {
+			tC.log = logging.New(ioutil.Discard, 0)
+		}
+		if tC.res == nil && !tC.noResolver {
+			tC.res = resolverMock.NewResolver(
+				resolverMock.WithResolveFunc(func(string) (swarm.Address, error) {
+					return tC.wantAdr, nil
+				}))
+		}
+
+		s := api.New(nil, nil, tC.res, nil, tC.log, nil).(*api.Server)
```

Comment:
why the cast?

Reply:
Currently, the `server` struct is internal to the api packagem and the `ResolveNameOrAddress` is it's member. This is admittedly a patchwork solution, but a PR for refactoring is already open #637, which will solve this issue.

Reply:
i don't see what you mean. `api.New` returns the interface, not the unexported type:
`func New(tags *tags.Tags, storer storage.Storer, corsAllowedOrigins []string, logger logging.Logger, tracer *tracing.Tracer) Service`